### ISSUE TITLE
Every object `is` every type marked with `[ObjectLiteral]`

### ISSUE THREAD

**Issue by dionrhys** [bridgedotnet/Bridge#3164]
I'm not sure what can be done about this, or if it's come up before.

It appears that having an `ObjectLiteral` type on the right-hand side of the `is` operator causes Bridge to always report that an object is an instance of that `ObjectLiteral`-based type.

I'm guessing this would be hard to make work because type information is not kept for `ObjectLiteral` types. And in that case, maybe a compiler error would be better here instead, to prevent putting `ObjectLiteral` types on the right-hand side of the `is` operator.

Regardless, I don't think the current behaviour of every object passing the `is` check against an `ObjectLiteral` type is correct.

### Steps To Reproduce

https://deck.net/dfbb70defd28c871de904d8441c01aa5

```csharp
public class Program
{
    public static void Main()
    {
        string s = "hello";
        Console.WriteLine(s is Derp);
    }
    
    [ObjectLiteral]
    public class Derp
    {
        public int Foo { get; set; }
    }
}
```

### Expected Result

```
Compile error: Types marked with ObjectLiteral cannot be used on the right-hand side of the `is` operator.
```

### Actual Result

```js
Bridge.assembly("Demo", function ($asm, globals) {
    "use strict";

    Bridge.define("Demo.Program", {
        main: function Main () {
            var s = "hello";
            System.Console.WriteLine(System.Boolean.toString(Bridge.is(s, System.Object)));
        }
    });
});
```

**Output:**
```
True
```

**Comment by vladsch** [comment]
Yes, maybe silent ignoring of cast operation is not good idea. We will review it.
By the way, ObjectLiteral can support casting for ObjectCreateMode.Constructor

**Comment by fabriciomurta** [comment]
The associated pull request has just been merged in. The fix will make it to Bridge 16.4.0.
